python/port/mod/turtle/turtle.cpp

```diff
@@ -74,19 +74,22 @@ void Turtle::left(mp_float_t angle) {
 void Turtle::circle(mp_int_t radius, mp_float_t angle) {
   mp_float_t oldHeading = heading();
   mp_float_t length = std::fabs(angle * k_headingScale * radius);
-  if (length > 1) {
-    for (int i = 1; i < length; i++) {
-      mp_float_t progress = i / length;
-      // Move the turtle forward
-      if (forward(1)) {
-        // Keyboard interruption. Return now to let MicroPython process it.
-        return;
+  int direction = 1; // 1=fwd, -1=backward
+  if (angle < 0) {direction =-1;} 
```

Comment:
Can you add spaces ? (And remove `;`, if possible, I don't know if it can build without…)
```suggestion
  if (angle < 0) { direction =-1; } 
```

Reply:
Spaces removed (in next commit). 
; maintained